extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,105 @@
+import os.path
+
+from packageurl import PackageURL
+from cyclonedx.factory.license import LicenseFactory
+from cyclonedx.model import ExternalReference, ExternalReferenceType, LicenseChoice, XsUri
+from cyclonedx.model.bom import Bom
+from cyclonedx.model.component import Component, ComponentType
+from cyclonedx.output.json import JsonV1Dot4
+
+from conan.api.output import cli_out_write
+from conan.api.conan_api import ConanAPI
+from conan.cli.args import common_graph_args, validate_common_graph_args
+from conan.cli.command import conan_command
+from conans.client.graph.graph import Node
+
+lFac = LicenseFactory()
+
+
+def package_type_to_component_type(pt: str) -> ComponentType:
+    if pt is "application":
+        return ComponentType.APPLICATION
+    else:
+        return ComponentType.LIBRARY
+
+
+def licenses(ids):
+    """
+    see https://cyclonedx.org/docs/1.4/json/#components_items_licenses
+    """
+    if ids is None:
+        return None
+    if not isinstance(ids, tuple):
+        ids = [ids]
+    return [LicenseChoice(license=lFac.make_from_string(i)) for i in ids]
+
+
+def package_url(node: Node) -> PackageURL:
+    """
+    Creates a PURL following https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#conan
+    """
+    return PackageURL(
+        type="conan",
+        name=node.conanfile.name,
+        version=node.conanfile.version,
+        qualifiers={
+            "prev": node.prev,
+            "rref": node.ref.revision,
+            "user": node.conanfile.user,
+            "channel": node.conanfile.channel,
+            "repository_url": node.remote.url if node.remote else None
+        })
+
+
+def create_component(n: Node) -> Component:
+    result = Component(
+        type=package_type_to_component_type(n.conanfile.package_type),
+        name=n.conanfile.name,
+        version=n.conanfile.version,
+        licenses=licenses(n.conanfile.license),
+        bom_ref=package_url(n).to_string(),
+        purl=package_url(n),
+        description=n.conanfile.description
+    )
+    if n.conanfile.homepage:
+        result.external_references.add(ExternalReference(
+            type=ExternalReferenceType.WEBSITE,
+            url=XsUri(n.conanfile.homepage),
+        ))
+    return result
+
+
+@conan_command(group="Recipe")
+def create_sbom(conan_api: ConanAPI, parser, *args):
+    """
+    creates an SBOM in CycloneDX 1.4 JSON format from a Conan graph JSON
```

Comment:
The "graph JSON" part is outdated by now.

Reply:
dropped